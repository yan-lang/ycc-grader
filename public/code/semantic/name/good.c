int main()
{
    int a = 10;
    a = a + 1;
    int b = 20;

    {
        int a = 9;
        a = a + b;
    }
}