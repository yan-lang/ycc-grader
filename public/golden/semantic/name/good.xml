<nameResolve>
    <def>
        <line>1</line>
        <type>func</type>
        <name>main</name>
    </def>
    <def>
        <line>3</line>
        <type>var</type>
        <name>a</name>
    </def>
    <ref>
        <line>4</line>
        <type>var</type>
        <name>a</name>
        <refLine>3</refLine>
    </ref>
    <ref>
        <line>4</line>
        <type>var</type>
        <name>a</name>
        <refLine>3</refLine>
    </ref>
    <def>
        <line>5</line>
        <type>var</type>
        <name>b</name>
    </def>
    <def>
        <line>8</line>
        <type>var</type>
        <name>a</name>
    </def>
    <ref>
        <line>9</line>
        <type>var</type>
        <name>a</name>
        <refLine>8</refLine>
    </ref>
    <ref>
        <line>9</line>
        <type>var</type>
        <name>a</name>
        <refLine>8</refLine>
    </ref>
    <ref>
        <line>9</line>
        <type>var</type>
        <name>b</name>
        <refLine>5</refLine>
    </ref>
</nameResolve>